### ISSUE TITLE
Allow to disable "Priority Inbox" (or set another starting inbox)

### ISSUE THREAD

**Issue by rnwgnr** [nextcloud/mail#3265]
### Feature Request
WIth release 1.4.0 the mail app introduces the "Priority Inbox" which is set as the default inbox when opening the mail app.

I'd like to have a setting to disable the "Priority Inbox" completely or at least set a another inbox (especially the "shared inbox" like in previous app version) as the default inbox to open.

**Comment by jasond2020** [comment]
In this context it would be great, if there were some preferences for choosing what is shown when "Priority Inbox" is selected. Status Quo: Priority Inbox, Favorites, Inbox. It would be good to be able to select if one only wants the see Priority Inox and Inbox. In my opinion the 'Favorites' are more of long-term interest and should not be visible all the time.

**Comment by guzzisti** [comment]
@jasond2020 Please open another issue for your request. This is something different than what i requested.

**Comment by dcrobertson01** [comment]
I would like this too. I don't know how it is supposed to work but all I get is the same emails sitting in the 'Priority Inbox' for days on end and another one occasionally joins them. I can't see any logic behind which ones are 'priority' and which ones are not.

**Comment by ChristophWurst** [comment]
I think what would make this feature more useful is to show *important and unread* in the first section and not all *important* messages, otherwise the sit there too long indeed.

**Comment by ChristophWurst** [comment]
>  I can't see any logic behind which ones are 'priority' and which ones are not.

It's a bit complicated to explain but basically the app looks at your other email and tries to see patterns in what is important to you, like senders from which you read most email, or people who you are in contact the most. Then it will classify the incoming messages with this knowledge.

https://nextcloud.com/blog/nextcloud-mail-introduces-machine-learning-for-priority-inbox/ might shed a bit of light on this black box.

**Comment by jasond2020** [comment]
Yes, but...
This seems to be not really working (at least for me), because there is no causality between 'many' and important (many... ...replies, ...mails, ...quotes, or whatever you have thought of for your algorithm).
My experience with this 'feature' is: it does nothing, except if i mark one specific mail as important; after that the mail stays up
I would like to see something like this: Mark a sender (or something within the mail) als important and see every new mail fitting this 'rule' be marked important automatically. If I unmark this 'rule' (for whatever reason) i want to be asked, if this only applies to new mails or to old mails too (which then would be unmarked as well). This way there would be two (three) ways of handling the inbox: 1st: mark as favorite (static), 2nd: mark as importand (dynamically, rule-based), (3rd: not marked, 'ornary' mails)

As for now: this feature is not helping, it only makes the view on my mobile less practical, because the important and favorits are always showing up und for the new mails i have to scroll...

And for the Priority of this topic: this is high in so far, as there has to be a way to disable this (at the moment anti-)feature

**Comment by ChristophWurst** [comment]
> I would like to see something like this: Mark a sender (or something within the mail) als important and see every new mail fitting this 'rule' be marked important automatically. If I unmark this 'rule' (for whatever reason) i want to be asked, if this only applies to new mails or to old mails too (which then would be unmarked as well).

This works great for you. It will be terrible for other people's workflow. That's the issue with rules and why decided not to do rule-based classification.

> And for the Priority of this topic: this is high in so far, as there has to be a way to disable this (at the moment anti-)feature

I'm the maintainer and I have to set priorities. Right now it's more important to improve the feature than to prematurely disable it. If you're interested in making the feature better we can discuss ways to debug it. Just stating that it doesn't work does not help.

You're free to invest time into improving the code and send pull requests.

**Comment by jasond2020** [comment]
> > I would like to see something like this: Mark a sender (or something within the mail) als important and see every new mail fitting this 'rule' be marked important automatically. If I unmark this 'rule' (for whatever reason) i want to be asked, if this only applies to new mails or to old mails too (which then would be unmarked as well).
> 
> This works great for you. It will be terrible for other people's workflow. That's the issue with rules and why decided not to do rule-based classification.
> 
> > And for the Priority of this topic: this is high in so far, as there has to be a way to disable this (at the moment anti-)feature
> 
> I'm the maintainer and I have to set priorities. Right now it's more important to improve the feature than to prematurely disable it. If you're interested in making the feature better we can discuss ways to debug it. Just stating that it doesn't work does not help.
> 
> You're free to invest time into improving the code and send pull requests.

You are absolutely right; my intention was to point out that i really would like to see this working in a way that fits the expectations... To me it seems like a good way, to make this an optional feature, so it can be improved while at the same time giving the user the chance to disable it as long as it does not fit the (specific) expectations. And you are right in another way: It is not my place to 'demand' something as long as i don't write the code; so, sorry for the harsh tone!

**Comment by ChristophWurst** [comment]
All good :v: 

I'll think about how we can debug the priority inbox better. It's often hard to describe why it feels like it does not work. And then we can't reproduce anyhting. Maybe there is away to extract the predictions.

**Comment by Karamelmar** [comment]
my request would also be, to have an option to disable the "Priority Inbox" and also "All Inboxes" or maybe an option to define a default view inbox/folder would be an alternative?
Actually i like the Priority inbox but since some users of our nextcloud instance are quit confused by the automated sorting, it would be nice to have the opportunity to choose.

**Comment by ogruenig** [comment]
I would also like this "feature", and it would be good to have it soon as I start forgetting new emails.
Of course the feature of a well functioning "priority inbox" sounds compelling, but unfortunately it doesn't work for me at this time (and apparently most users), so in my "production environment" I would like to disable this until it is useful.

**Comment by umrath** [comment]
As a user my primary interest is a working app.
This is not the case.
So, my only option is to abandon the usage because the maintainer prefers to try to fix the feature instead of allowing users to disable it, when they do not want to have it.

I use rainloop now. It's not perfect either. But it has no bugs that compromise usability.

**Comment by LukaPitamic** [comment]
+1 for option to remove priority inbox. 

As central communication point of "open platform" - officially endorsed and integrated email client in Nextcloud - app should definitely allow user to set it up the way he/she prefers. 
BY DEFINITION! 

Me personally(and I'm not very special), I set filters outside client to segment emails and throw them in dedicated "IMAP" folders.

Nevertheless - I LOVE NEXTCLOUD!!!!!!

**Comment by Ornias1993** [comment]
Agreed, me and my wife tried it and don't want it.
Why do we need to be forced to have a priority inbox that could quite easily be disabled?

**Comment by Ornias1993** [comment]
Okey i've looked into it a bit, it seems we can change the mailbox on load with in this code portion:

```
		if (this.$route.name === 'home' && accounts.length > 1) {
			// Show first account
			const firstAccount = accounts[0]
			// FIXME: this assumes that there's at least one mailbox
			const firstMailbox = this.$store.getters.getMailboxes(firstAccount.id)[0]
			console.debug('loading first mailbox of first account', firstAccount.id, firstMailbox.databaseId)

```
in src/views/Home.vue

Also we can just hide the menu item using CSS:
`id="mailbox-priority"`

We just need a toggle to set both and we're done.

**Comment by LukaPitamic** [comment]
@Ornias1993 elegant solution - css override! Does it focus automatically on inbox when mail app is loaded?

**Comment by Ornias1993** [comment]
> @Ornias1993 elegant solution - css override! Does it focus automatically on inbox when mail app is loaded?

Yes, but you still need to replace the first mailbox that loads when you click "mail", I think that can be done with the earlier mentioned codeportion.

**Comment by ChristophWurst** [comment]
Okay, I have some ideas. Could everyone with issues please help us debug those problems? I've started a separate ticket at https://github.com/nextcloud/mail/issues/3968. Your help is really important. I can't debug those problems as long as they are not reproducible locally …

**Comment by ChristophWurst** [comment]
> To be fair to @ChristophWurst I never like(d) priority mailboxes in any way-shape or form. It has nothing to do with your code being bad. Some people might simply not like the whole concept.... call it "old fasioned" ;-)

Understandable :v:

**Comment by ChristophWurst** [comment]
Reminder that I'm waiting for feedback with the priority inbox problems at https://github.com/nextcloud/mail/issues/3968 :v:

**Comment by cyrilst** [comment]
> > To be fair to @ChristophWurst I never like(d) priority mailboxes in any way-shape or form. It has nothing to do with your code being bad. Some people might simply not like the whole concept.... call it "old fasioned" ;-)
> 
> Understandable ✌️

Can I insist a bit more on this ? While the PI is an awesome functionality, some people are really uncomfortable with it. There is an option to unsubscribe from every folder. I'm sure something similar for the PI would be perfect. Also, completely disabling the feature would surely save some resources…

**Comment by ChristophWurst** [comment]
Right now it's more a question of having enough people to test this additional path in our application than a technical limitation.

The more testers, reviewers and contributors we have, the easier. If you wish to help make this app better with one of those three contributions please me know.

**Comment by feutl** [comment]
@ChristophWurst 
I am a little lost, too many references. Anyhow, how can we / I help to test whatever path ;)

**Comment by umrath** [comment]
I'm not sure if you get the testing you wish for from the people interested in this fix here.
At least I abandoned the mail app for the time being (and use rainloop for now) until I get the information that Priority Inbox is gone.

Of course I cannot speak for the others. But I wouldn't be surprised if the others would react in a similar way.

**Comment by ChristophWurst** [comment]
Keep trolling, I don't care if you use a different email client.

**Comment by ChristophWurst** [comment]
> @ChristophWurst
> I am a little lost, too many references. Anyhow, how can we / I help to test whatever path ;)

It's not about this change exclusively, it's about testing the app (prereleases) on a regular basis to ensure we don't break anything.

**Comment by umrath** [comment]
> Keep trolling, I don't care if you use a different email client.

Why do you think I'm trolling.
I'm merely trying to explain why you do not get the test coverage you wish to get.
You drive people away with your attitude.

This issue is serious. Please handle it as such.

**Comment by ChristophWurst** [comment]
Just because I don't follow what you request doesn't mean I don't take this serious. You might want to read up on https://nextcloud.com/contribute/code-of-conduct/.

**Comment by ChristophWurst** [comment]
For those that are here to constructively move this forward, here's some info for the required change
* PI only exists client-side, hence no back-end needs to be changed
* It's all handled in https://github.com/nextcloud/mail/tree/master/src/store
* There is lost of existing tests that ensure the integrity of the PI logic, those need to be extended for this, see https://github.com/nextcloud/mail/tree/master/src/tests/unit/store
* Background synchronization has to be rewritten for both parts as the current logic is tailored for PI updates

Happy tinkering :v:

**Comment by dodedodo** [comment]
I understand the urge to fix Priority Inbox rather than disable it. 

However, I do think some users are never going to be comfortable with it. Simply because 'it does magical things', and they are afraid they'll miss an important email. Myself included. For some users, this request has nothing to do with the actual functionality of PI. 

---

@ChristophWurst If PI only exists client-side, how do you guarantee the same PI view across browsers/session/platform? Surely there has to be some server-side storage involved?

**Comment by LukaPitamic** [comment]
@dodedodo I totally agree. I do my own prioritization.

**Comment by Ornias1993** [comment]
> However, I do think some users are never going to be comfortable with it. Simply because 'it does magical things', and they are afraid they'll miss an important email. Myself included. For some users, this request has nothing to do with the actual functionality of PI.

For me personally it has nothing to do with the magical things, but more the fact I do most things in my workflow semi-blind. Simply put: I know expected email/button/function is at spot x. Intelligent inboxes (no mater how great) are not very great fit for said workflow.

**Comment by xysmalobia** [comment]
> For some users, this request has nothing to do with the actual functionality of PI.

Same here. In fact I stopped using Gmail because of its priority inboxes and switched to nextcloud exactly because it didn't tamper with my emails. I use a lot of different folders to organise my mail. So far priority inbox has also caused me to miss important emails that came from new contacts. I'd find it extremely helpful if we could disable this feature.

**Comment by freekvh** [comment]
Just adding my 2ct. I love this mail plugin, it is my primary mail client. I can also understand that some people like a priority inbox. But I actually take great care that I only receive emails with priority (unsubscribe from everything, label spam as spam, block addresses). As a result I think the priority email feature does not work for me. Actually it seems to even be reversed, because of this "filter" I notice email too late. It would be nice if my normal inbox was the default view. Also the widget has been showing the same couple for "important" emails for a long time now, hiding actually important new emails from me.

I would also really appreciate being able to turn this off.

Thanx and keep up the great work!

**Comment by ChristophWurst** [comment]
> Also the widget has been showing the same couple of "important" emails for a long time now, hiding actually important new emails from me.

That is fixed with v1.8 FYI.

**Comment by freekvh** [comment]
> 
> 
> > Also the widget has been showing the same couple of "important" emails for a long time now, hiding actually important new emails from me.
> 
> That is fixed with v1.8 FYI.

I have 1.8.0, still it does not display every unread mail in the widget. I think the volume of mail I receive is just too low and so the number of emails marked "important" is even lower (in the order of one per several days). In reality all my emails are important to me and I can very easily give attention to every single email. If an email is not important to me, I take actions to make sure I don't receive similar mail in the future anymore.

Maybe I would appreciate the system more when I would receive more email. For now though, I feel like 95% of emails that is not labeled as important is still important to me. So I'd love to be able to turn this off. And I love for the widget to just let me see my most recent and unread mails in chronological order.

**Comment by kilrah** [comment]
+1 for being able to disable it.

> > I would like to see something like this: Mark a sender (or something within the mail) als important and see every new mail fitting this 'rule' be marked important automatically. If I unmark this 'rule' (for whatever reason) i want to be asked, if this only applies to new mails or to old mails too (which then would be unmarked as well).
> 
> This works great for you. It will be terrible for other people's workflow. That's the issue with rules and why decided not to do rule-based classification.

But with rules you can at least set them how you want them and fix things when they're wrong, an automated thing is bound to get things wrong potentially more often and you can't do anything about it, IMO making it an anti-feature as you spend more time assessing whether the automated thing did a good job or not rather than triaging stuff manually in the first place. I've got no problem with people liking it, but for those who don't it should be possible to disable.

> Right now it's more important to improve the feature than to prematurely disable it. If you're interested in making the feature better we can discuss ways to debug it. 

Personally I'm not for using such automated features unless they're basically 100% accurate all the time, and I don't know of ways it could be.

Currently for me it causes issues ironically not because it's wrong, it actually picks an "important and unread" email as the first to show, except that has the side effect that "enter mail page -> app selects priority inbox -> selects first mail in there -> that mail is now marked as read automatically", which I precisely didn't want.

**Comment by Ornias1993** [comment]
> > Right now it's more important to improve the feature than to prematurely disable it. If you're interested in making the feature better we can discuss ways to debug it.
> 
> Personally I'm not for using such automated features unless they're basically 100% accurate all the time, and I don't know of ways it could be.

Even if it would be, I don't get why it needs to be forced on users and it's to much to add a toggle for the maintainer. 

I understand why this issue isn't high priority, I myself have community requirested issues open, while I rather took on some structural reworks first. But at least be fair and honest, just pointing out you're only interested in improving the feature showcases a big disregard for your community.

**Comment by ChristophWurst** [comment]
I summarized the required changes at https://github.com/nextcloud/mail/issues/3265#issuecomment-766637799. Happy hacking.

**Comment by kilrah** [comment]
> Just because I don't follow what you request doesn't mean I don't take this serious. You might want to read up on https://nextcloud.com/contribute/code-of-conduct/.

Interesting that you would point to the code of conduct, the very first point of it being 

> Be considerate
Your actions and work will affect and be used by other people and you in turn will depend on the work and actions of others. Any decision you take will affect other community members, and we expect you to take those consequences into account when making decisions.

The decision to shove a feature that affects people's workflows down their throat without a way to disable it and telling them to go hack the code if they don't want it seems like exactly the kind of thing that violates that point of the code of conduct.

**Comment by freekvh** [comment]
> 
> 
> > Just because I don't follow what you request doesn't mean I don't take this serious. You might want to read up on https://nextcloud.com/contribute/code-of-conduct/.
> 
> Interesting that you would point to the code of conduct, the very first point of it being
> 
> > Be considerate
> > Your actions and work will affect and be used by other people and you in turn will depend on the work and actions of others. Any decision you take will affect other community members, and we expect you to take those consequences into account when making decisions.
> 
> The decision to shove a feature that affects people's workflows down their throat without a way to disable it and telling them to go hack the code if they don't want it seems like exactly the kind of thing that violates that point of the code of conduct.

I think you need to stop with these kinds of remarks. As user of a free product you are entitled to exactly nothing. And yet you get a product and even the opportunity to change it and add functionality to it, if you want. You should be nothing but grateful.

Frankly I feel that many foss devs poor a lot of passion into a lot of projects and that passion is often fueled by curiosity and the wish to learn about (new) subjects. In this case it may be the wish to learn about AI powered filtering. Other things may feel like adjacencies. One can work outside of their current field of passion but I for one, would require payment for that.

You have to be careful with you critical attitude because it can be the reason foss devs give up on their passion project because there is just too much negativity, like in this thread.

So just do it yourself and don't complain. Or just respectfully file a feature request and leave the rest up to the dev, and don't complain.

**Comment by kilrah** [comment]
I am a FOSS dev myself, and completely understand the point.

> and that passion is often fueled by curiosity and the wish to learn about (new) subjects. In this case it may be the wish to learn about AI powered filtering

Sure, nothing against having that as an option. But forcing a feature that should be "accessory" and doesn't impact the core functionality / have a good reason to be forced other than "I worked on it so you'll have no choice but using it now and be happy about it" when there's user backlash is doing no one any good and is really questionalble as a dev attitude. You can learn about anything that motivates you and make something out of it, but that doesn't mean it's something people will necessarily need/want so if some turn out not to want it you make it optional, easy.

**Comment by jancborchardt** [comment]
Further discussion here does not help, so I’ll lock this issue.

- If you want to help with implementing the ability to disable the priority inbox, see https://github.com/nextcloud/mail/issues/3265#issuecomment-766637799
- If you want to help improve the priority inbox so it’s more useful, please help by submitting info at https://github.com/nextcloud/mail/issues/3968

About the tone of writing, also echoing what @freekvh said above: We all work well in a friendly environment – if there’s a lot of angry criticism on something going _beyond_ the necessary info to debug it or help out, then it’s not only unproductive, but counterproductive as it’s demotivating any contributor. This will lead to the issue in question being fixed _much slower_ instead of faster.

So let’s please keep it extra friendly going forward. Thanks! :slightly_smiling_face:

**Comment by jancborchardt** [comment]
The original issue was about the problem that the Priority inbox _always_ opens by default when opening the Mail app. We fixed this some time ago by always remembering which folder was open last, and opening that on load. This is a very nice and universal solution, so we consider this issue fixed. :)
